### ISSUE TITLE
Bump `actions/checkout` to `@v4` and `actions/github-script` to `@v7` to get rid of the node16 warnings.

### ISSUE THREAD

**Issue by ianspektor** [google/temporian#388]
Fwiw, once this is merged, someone should go in and bump `actions/checkout` to `@v4` and `actions/github-script` to `@v7` to get rid of the node16 warnings.

I'm leaving that for a separate PR as I don't really want to extend this PR -- it's already sufficiently complicated.

_Originally posted by @jsoref in https://github.com/google/temporian/issues/385#issuecomment-1986170556_
